### ISSUE TITLE
Domoticz log errors

### ISSUE THREAD

**Issue by MaxX850** [MFxMF/SDM630-Modbus#2]
Hello,
Domoticz v4.9700, rpi zero w. I have errors like this : 

- 2019-01-22 14:20:38.146 Error: (Eastron SDM630) ----> Line 552 in /home/pi/domoticz/plugins/SDM630-Modbus/plugin.py, function onHeartbeat
- 2019-01-22 14:20:38.146 Error: (Eastron SDM630) ----> Line 386 in /home/pi/domoticz/plugins/SDM630-Modbus/plugin.py, function onHeartbeat
- 2019-01-22 14:20:48.086 Error: (Eastron SDM630) 'onHeartbeat' failed 'KeyError'.

For test I connect only 1 phase. I see voltage, current (only max demand), power, power factor, etc., but domoticz log is full of errors. Also, can not send these eastron sensors throu master-slave link beetwen two servers with domoticz (other sensors like ds18b20, rpi internal sensors are sended without any problems). Python-pip, pyserial, minimalmodbus installed. Newest raspbian. When I switch debug mode to true, errors stop appearing but link between domoticzes is still not sending data from eastron.

**Comment by art0** [comment]
Thanks for the plugin :)
Any updates on this issue?
 
2019-02-27 13:00:49.706  Error: (Power0) 'onHeartbeat' failed 'AttributeError':''str' object has no attribute 'read_float''.
2019-02-27 13:00:49.706  Error: (Power0) ----> Line 552 in '/var/tmp/domoticz/plugins/SDM630-Modbus/plugin.py', function onHeartbeat
2019-02-27 13:00:49.706  Error: (Power0) ----> Line 265 in '/var/tmp/domoticz/plugins/SDM630-Modbus/plugin.py', function onHeartbeat
2019-02-27 13:00:49.706  Error: (Power0) 'onHeartbeat' failed 'AttributeError':''str' object has no attribute 'read_float''.
2019-02-27 13:00:49.706  Error: (Power0) ----> Line 552 in '/var/tmp/domoticz/plugins/SDM630-Modbus/plugin.py', function onHeartbeat
2019-02-27 13:00:49.706  Error: (Power0) ----> Line 265 in '/var/tmp/domoticz/plugins/SDM630-Modbus/plugin.py', function onHeartbeat
2019-02-27 13:00:49.806  Error: (Power1-DC) 'onHeartbeat' failed 'AttributeError':''str' object has no attribute 'read_float''.

**Comment by dentichel** [comment]
Similar story here... Installed and activated the plugin successfully under Domoticz 4.10603. 

However the execution of the plugin fails with errors like: 
2019-04-17 14:05:21.728 Error: ( SDM630) 'onHeartbeat' failed 'OSError'.
2019-04-17 14:05:21.728 Error: ( SDM630) ----> Line 552 in '/home/pi/domoticz/plugins/SDM630-Modbus/plugin.py', function onHeartbeat
2019-04-17 14:05:21.729 Error: ( SDM630) ----> Line 265 in '/home/pi/domoticz/plugins/SDM630-Modbus/plugin.py', function onHeartbeat
2019-04-17 14:05:21.729 Error: ( SDM630) ----> Line 392 in '/home/pi/domoticz/plugins/SDM630-Modbus/minimalmodbus.py', function read_float
2019-04-17 14:05:21.729 Error: ( SDM630) ----> Line 697 in '/home/pi/domoticz/plugins/SDM630-Modbus/minimalmodbus.py', function _genericCommand
2019-04-17 14:05:21.729 Error: ( SDM630) ----> Line 795 in '/home/pi/domoticz/plugins/SDM630-Modbus/minimalmodbus.py', function _performCommand
2019-04-17 14:05:21.729 Error: ( SDM630) ----> Line 930 in '/home/pi/domoticz/plugins/SDM630-Modbus/minimalmodbus.py', function _communicate

As an FYI: I have set all of the .py files as executable (chmod +x) and restarted domoticz, but to no avail. Furthermore, I have tried with the installation of pyserial, minimalmodbus - and 'serial' as an alternative. Both under python2 and python3.
